src/modules/mqtt/README

```diff
@@ -0,0 +1,349 @@
+
+   Example 1.9. Set event_callback parameter
+...
+modparam("mqtt", "event_callback", "ksr_mqtt_event")
+...
+-- event callback function implemented in Lua
+function ksr_mqtt_event(evname)
+        KSR.info("===== mqtt module triggered event: " .. evname .. "\n");
+        return 1;
+end
+
+-- event callback function implemented in Python
+function ksr_mqtt_event(self, msg, evname)
+        KSR.info("===== mqtt module triggered event: %s\n" % evname);
+        return 1;
+end
+...
+
+4. Functions
+
+   4.1. mqtt_subscribe(topic, qos)
+   4.2. mqtt_unsubscribe(topic)
+   4.3. mqtt_publish(topic, message, qos)
+
+4.1.  mqtt_subscribe(topic, qos)
+
+   Subscribe to the given topic. Mqtt qos levels 0, 1 and 2 can be used.
+
+   The function is passing the task to mqtt dispatcher process, therefore
+   the SIP worker process is not blocked.
+
+   Incoming messages for this topic are then handled by the same process
+   and exposed to the event_route[mqtt:message].
+
+   This function can be used from ANY_ROUTE.
+
+   Example 1.10. mqtt_subscribe usage
+...
+mqtt_subscribe("kamailio/script", 0);
+...
+
+4.2.  mqtt_unsubscribe(topic)
+
+   Unsubscribe to a previously subscribed topic. The mqtt broker will stop
+   forwarding messages for this topic.
+
+   This function can be used from ANY_ROUTE.
+
+   Example 1.11. mqtt_unsubscribe usage
+...
+mqtt_unsubscribe("kamailio/script");
+...
+
+4.3.  mqtt_publish(topic, message, qos)
+
+   Send out a message to a topic with a specified mqtt qos level (0, 1,
+   2). Again the actual sending is done in a mqtt dispatcher process and
+   will not block the SIP worker.
+
+   Example 1.12. mqtt_publish usage
+...
+mqtt_publish("kamailio/event", "some message", 0);
+...
+
+5. Event routes
+
+   5.1. mqtt:connected
+   5.2. mqtt:disconnected
+   5.3. mqtt:message
+
+5.1.  mqtt:connected
+
+   If defined, the module calls event_route[mqtt:connected] when a
+   outgoing broker connection is established.
+
+   MQTT subscriptions are not durable, so you should use this event route
+   to manage your subscriptions.
+...
+event_route[mqtt:connected] {
+        xlog("mqtt connected !\n");
+        mqtt_subscribe("kamailio/script", 0);
+}
+...
+
+5.2.  mqtt:disconnected
+
+   If defined, the module calls event_route[mqtt:disconnected] when the
+   broker connection is lost.
+
+   The module will automatically try to reconnect to the broker every 3
+   seconds.
+...
+event_route[mqtt:disconnected] {
+    xlog("Lost mqtt connection !\n");
+}
+...
+
+5.3.  mqtt:message
+
+   If defined, the module calls event_route[mqtt:message] when a message
+   is received from the broker.
+
+   All incoming messages are handled in a single mqtt dispatcher process.
+...
+event_route[mqtt:message] {
+        xlog("mqtt message received: $mqtt(topic) -> $mqtt(msg) !\n");
+}
+...
+
+6. Exported pseudo-variables
+
+     * $mqtt(topic) - Received topic (only in mqtt:message)
+     * $mqtt(msg) - Received message (only in mqtt:message)
+
```

Comment:
How do I reach the QoS value for the message?

Reply:
@oej  it's not exposed ATM but it's simple to add it. I will enhance the module accordingly.

Reply:
@oej, see commit ea69941